src/ray/core_worker/task_manager.h

```diff
@@ -87,6 +87,51 @@ using PushErrorCallback = std::function<Status(const JobID &job_id,
                                                const std::string &error_message,
                                                double timestamp)>;
 
+/// When the streaming generator tasks are submitted,
+/// the intermediate return objects are streamed
+/// back to the task manager.
+/// This class manages the references of intermediately
+/// streamed object references.
+/// The API is not thread-safe.
+class ObjectRefStream {
+ public:
+  ObjectRefStream(const ObjectID &generator_id) : generator_id_(generator_id) {}
+
+  /// Asynchronously read object reference of the next index.
+  ///
+  /// \param[out] object_id_out The next object ID from the stream.
+  /// Nil ID is returned if the next index hasn't been written.
+  /// \return KeyError if it reaches to EoF. Ok otherwise.
+  Status AsyncReadNext(ObjectID *object_id_out);
```

Comment:
TryReadNextItem? the 'Async` is a bit confusing here.

Reply:
sgtm